mongodbatlas/fw_resource_mongodbatlas_alert_configuration.go

```diff
@@ -0,0 +1,811 @@
+						},
+						"victor_ops_routing_key": schema.StringAttribute{
+							Sensitive: true,
+							Optional:  true,
+						},
+						"roles": schema.ListAttribute{
+							ElementType: types.StringType,
+							Optional:    true,
+						},
+						"microsoft_teams_webhook_url": schema.StringAttribute{
+							Sensitive: true,
+							Optional:  true,
+						},
+						"webhook_secret": schema.StringAttribute{
+							Sensitive: true,
+							Optional:  true,
+						},
+						"webhook_url": schema.StringAttribute{
+							Sensitive: true,
+							Optional:  true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *AlertConfigurationRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.client.Atlas
+
+	var alertConfigPlan tfAlertConfigurationRSModel
+
+	diags := req.Plan.Get(ctx, &alertConfigPlan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := alertConfigPlan.ProjectID.ValueString()
+
+	apiReq := &matlas.AlertConfiguration{
+		EventTypeName:   alertConfigPlan.EventType.ValueString(),
+		Enabled:         alertConfigPlan.Enabled.ValueBoolPointer(),
+		Matchers:        expandTFAlertConfigurationMatchers(alertConfigPlan.Matcher),
+		MetricThreshold: expandTFAlertConfigurationMetricThresholdConfig(alertConfigPlan.MetricThresholdConfig),
+		Threshold:       expandTFAlertConfigurationThresholdConfig(alertConfigPlan.ThresholdConfig),
+	}
+
+	notifications, err := expandTFAlertConfigurationNotification(alertConfigPlan.Notification)
+	if err != nil {
+		resp.Diagnostics.AddError(errorCreateAlertConf, err.Error())
+		return
+	}
+	apiReq.Notifications = notifications
+
+	apiResp, _, err := conn.AlertConfigurations.Create(ctx, projectID, apiReq)
+	if err != nil {
+		resp.Diagnostics.AddError(errorCreateAlertConf, err.Error())
+		return
+	}
+
+	encodedID := encodeStateID(map[string]string{
+		encodedIDKeyAlertID:   apiResp.ID,
+		encodedIDKeyProjectID: projectID,
+	})
+	alertConfigPlan.ID = types.StringValue(encodedID)
+
+	newAlertConfigurationState := toTFAlertConfigurationModel(apiResp, &alertConfigPlan)
+
+	// set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, newAlertConfigurationState)...)
+}
+
+func (r *AlertConfigurationRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	conn := r.client.Atlas
+
+	var alertConfigState tfAlertConfigurationRSModel
+
+	// get current state
+	resp.Diagnostics.Append(req.State.Get(ctx, &alertConfigState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ids := decodeStateID(alertConfigState.ID.ValueString())
+
+	alert, getResp, err := conn.AlertConfigurations.GetAnAlertConfig(context.Background(), ids[encodedIDKeyProjectID], ids[encodedIDKeyAlertID])
+	if err != nil {
+		// deleted in the backend case
+		if getResp != nil && getResp.StatusCode == http.StatusNotFound {
+			alertConfigState.ID = types.StringValue("") // TODO verify this works
```

Comment:
need to verify this works the same why it does in sdk v2 for marking resource as destroyed

Reply:
fixed to correct usage of remove resource method.